gettsim/pre_processing/piecewise_functions.py

```diff
@@ -24,22 +28,48 @@ def piecewise_linear(
 
 
     """
+
+    # Check if value lies within the defined range.
     if (value < lower_thresholds[0]) or (value > upper_thresholds[-1]):
-        out = np.nan
-    else:
-        index_interval = np.searchsorted(upper_thresholds, value, side="left")
-        if rates_modified:
-            # Calculate new intercept
-            intercept_interval = 0
-            for interval in range(index_interval):
-                intercept_interval += rates[interval] * (
+        return np.nan
+
+    index_interval = np.searchsorted(upper_thresholds, value, side="left")
+    if rates_modified:
+        # Calculate new intercept
+        intercept_interval = 0
+        for interval in range(index_interval):
+            for pol in range(1, rates.shape[0] + 1):
+                intercept_interval += (rates[pol - 1, interval] ** pol) * (
                     upper_thresholds[interval] - lower_thresholds[interval]
                 )
 
-        else:
-            intercept_interval = intercepts_at_lower_thresholds[index_interval]
+    else:
+        intercept_interval = intercepts_at_lower_thresholds[index_interval]
 
-        lower_thresehold_interval = lower_thresholds[index_interval]
-        rate_interval = rates[index_interval]
-        out = intercept_interval + (value - lower_thresehold_interval) * rate_interval
+    # Select threshold and calculate corresponding increment into interval
+    lower_thresehold_interval = lower_thresholds[index_interval]
+    increment_to_calc = value - lower_thresehold_interval
+
+    out = intercept_interval
+    for pol in range(1, rates.shape[0] + 1):
+        out += rates[pol - 1, index_interval] * (increment_to_calc ** pol)
     return out
+
+
+def add_progressionsvorbehalt(param_dict, parameter):
+    """
+    If no quadratic rate is given, add rate according to the
+    progressionsvorbehalt formula.
+    """
```

Comment:
Please extend the comment on how you obtain the "quadratic rate". what is the quadratic rate after all, i.e. what is the general functional form? [Progressionsvorbehalt](https://de.wikipedia.org/wiki/Progressionsvorbehalt) means to apply the tax tariff to two different income components and apply the tax rate from the second calculation to the first income concept. I don't think this is what happens here so we should use a diffferent name.

Reply:
Sorry I misunderstood the concept. Will change!

Reply:
Changed explanation and name to Progressionsfaktor.

Reply:
It's still not clear to me what this Progressionsfaktor actually is (I have a faint idea because I know the tax tariff, but this is impossible to understand without deeper knowledge.) Am I right that you infer the missing parameter by imposing equal tax dues from "below" and "above" each treshold?